File: cpp solution/maximum occuring character.cpp
class Solution
{
public:
    //Function to find the maximum occurring character in a string.
    char getMaxOccuringChar(string str)
    {
        // Your code here
        vector<int> freq(26, 0);
        for (char ch : str) {
            int number = 0;
            if (ch >= 'a' && ch <= 'z') {
                number = ch - 'a';
            }
            else {
                number = ch - 'A';
            }
            freq[number] += 1;
        }
        int _max = 0;
        int ans = 0;
        for (int i = 0; i < 26; i++) {
            if (freq[i] > _max) {
              // this will make sure that we get the lowest value character
                ans = i;
                _max = freq[i];
            }
        }
        return ans + 'a';
    }

};

File: python solution/maximum occuring character.py
# time: O(n) and space: O(1)
class Solution:
    # Function to find the maximum occurring character in a string.
    def getMaxOccurringChar(self, s):
        frequency = [0] * 26
        for char in s:
            char_number = 0
            if 'a' <= char <= 'z':
                char_number = ord(char) - ord('a')
            else:
                char_number = ord(char) - ord('A')
            frequency[char_number] += 1
        _max = 0
        char = 0
        for i in range(26):
            if frequency[i] > _max:
                char = i
                _max = frequency[i]
          # a = 97
        return chr(97 + char)
